File: API/Entities/AssetAggregate/Asset.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities.AssetAggregate
{
    public class Asset
    {
        [Key]
        public string Id { get; set; }
        public int No { get; set; }
        public string Name { get; set; }
        public string Type { get; set; }
        public string SerialNo { get; set; }

        /// <summary>
        /// Id 1 = "ทรัพย์สินของ บริษัท", Id 2 = "ทรัพย์สินของ คู่สัญญา"
        /// </summary>
        [ForeignKey("OwnerId")]
        public Owner Owner { get; set; }
        public int OwnerId { get; set; }

        public string Manufacturer { get; set; }
        public string Model { get; set; }
        public AssetStatus AssetStatus { get; set; } = AssetStatus.GoodCondition;

        [ForeignKey("StockId")]
        public Stock Stock { get; set; }
        public int StockId { get; set; }

        [ForeignKey("PersonInChargeId")]
        public int PersonInChargeId { get; set; }
        public User PersonInCharge { get; set; }

        public string AssetPic { get; set; }
        public string PublicId { get; set; }
    }
}

File: API/client/src/app/models/asset.ts
export interface Asset {
  id: string;
  no: number;
  name: string;
  type: string;
  serailNo: string;
  owner: Owner;
  manufacturer: string;
  model: string;
  assetStatus: string;
  stock: Stock;
  assetPic: string;
  personInCharge: PersonInCharge;
}

export interface Owner {
  id: number;
  ownerDesc: string;
}

export interface Stock {
  id: number;
  type: string;
}

export interface PersonInCharge {
  // New interface for personInCharge
  id: number;
  email: string;
  userName: string;
  position: string;
  department: string;
  section: string;
  phone: string;
  status: string;
}

interface AssetDetailsDto {
  id: string;
  assetId: string;
  assetPic: string;
  receivedDate: string;
  classifier: string;
  serialNo: string;
  supplier: string;
  totalAmount: number;
  vat: number;
  depreciationRate: number;
  grandAmount: number;
  usedMonths: number;
  department: string;
  section: string;
  locateAt: string;
  depreciation: number;
}

export interface AssetParams {
  orderBy: string;
  searchTerm?: string;
  pageNumber: number;
  pageSize: number;
}
